packages/bytecode-utils/src/lib/bytecode.ts

```diff
@@ -32,92 +55,291 @@ export const decode = (bytecode: string): DecodedObject => {
   }
 
   // split auxdata
-  const [, auxdata] = splitAuxdata(bytecode);
+  const [, auxdata] = splitAuxdata(bytecode, auxdataStyle);
+  // See more here: https://github.com/vyperlang/vyper/pull/3010
+  if (auxdataStyle === AuxdataStyle.VYPER) {
+    if (!auxdata) {
+      // Attempt fallback auxdata styles for older Vyper versions
+      const fallbackStyles = [
+        AuxdataStyle.VYPER_LT_0_3_10, // Vyper bytecode before version 0.3.10 uses SOLIDITY AuxdataStyle
+        AuxdataStyle.VYPER_LT_0_3_5, // Vyper bytecode before version 0.3.5 doesn't include auxdata length
+      ];
```

Comment:
I wonder what's the rationale of trying these fallbacks here? Wouldn't we call the `decode` function with the respective styles (`VYPER_LT_0_3_10`, ...) because we already know which version we want to verify for? 
I think I just need some explanation here.

Reply:
My rationale here is that when you call `decode(` you don't know the compiler version, you just try to decode a bytecode and you just know that is solidity or vyper.

While when you call `splitAuxdata` you know also the version. 

I think this is useful for services like `Monitor` in which you just try to decode a bytecode without knowing the version in advance.

Does this make sense? If yes we should definitely explain this in a comment or change the names of the functions